koku/masu/util/aws/common.py

```diff
@@ -499,9 +499,12 @@ def aws_generate_daily_data(data_frame):
     return daily_data_frame
 
 
-def match_openshift_resources_and_labels(data_frame, cluster_topology, matched_tags):
+def match_openshift_resources_and_labels(data_frame, cluster_topologies, matched_tags):
     """Filter a dataframe to the subset that matches an OpenShift source."""
-    resource_ids = tuple(cluster_topology.get("resource_ids", []))
+    resource_ids = []
+    for cluster_topology in cluster_topologies:
+        resource_ids.extend(cluster_topology.get("resource_ids", []))
```

Comment:
This is another more efficient technique for flattening a list of lists using [itertools.chain](https://docs.python.org/3/library/itertools.html#itertools.chain).
```suggestion
    resource_ids = itertools.chain.from_iterable(cluster_topology.get("resource_ids", []) for cluster_topology in cluster_topologies)
```

Reply:
I tried doing this with a comprehension using unpacking, but you can't do that. TIL. 🤓 
```
tuple(*cluster_topology.get("resource_ids", []) for cluster_topology in cluster_topologies)
  Cell In [74], line 1
    tuple(*cluster_topology.get("resource_ids", []) for cluster_topology in cluster_topologies)
          ^
SyntaxError: iterable unpacking cannot be used in comprehension
```

Reply:
implemented the change to using chain from itertools, good suggestion.

Reply:
It's neat, but I find that to be more taxing to reason through. More mental hoops. Just my two pennies.

Reply:
I agree it's a bit harder to read, but it's idiomatic Python (standard lib and a comprehension). Splitting over two lines may increase legibility.

```python
resource_ids = (cluster_topology.get("resource_ids", []) for cluster_topology in cluster_topologies)
resource_ids = itertools.chain.from_iterable(resource_ids)
```

I don't think this crosses over into the "clever" category, which is definitely something to be avoided.